diff --git a/3rdparty/CMakeLists.txt b/3rdparty/CMakeLists.txt
index 1f3d71b3..fec6ab2d 100644
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -58,8 +58,8 @@ else ()
 endif ()
 
 if (BUILD_GLEW)
-    add_subdirectory(glew)
-    INSTALL_HEADERS(glew)
+    #add_subdirectory(glew)
+    #INSTALL_HEADERS(glew)
 endif ()
 
 # GLFW
@@ -94,14 +94,14 @@ else ()
 endif ()
 
 if (BUILD_GLFW)
-    add_subdirectory(GLFW)
-    INSTALL_HEADERS(GLFW)
-    list(APPEND GLFW_INCLUDE_DIRS ${OPENGL_INCLUDE_DIR})
-    list(APPEND GLFW_LIBRARIES    ${OPENGL_gl_LIBRARY})
+    #add_subdirectory(GLFW)
+    #INSTALL_HEADERS(GLFW)
+    #list(APPEND GLFW_INCLUDE_DIRS ${OPENGL_INCLUDE_DIR})
+    #list(APPEND GLFW_LIBRARIES    ${OPENGL_gl_LIBRARY})
 endif ()
 
 # Azure Kinect
-include(azure_kinect/azure_kinect.cmake)
+#include(azure_kinect/azure_kinect.cmake)
 
 # JSONCPP
 if ((APPLE) AND ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU"))
@@ -166,8 +166,8 @@ else ()
 endif ()
 
 if (BUILD_JPEG)
-    message(STATUS "Building libjpeg-turbo from source")
-    include(libjpeg-turbo/libjpeg-turbo.cmake)
+    #message(STATUS "Building libjpeg-turbo from source")
+    #include(libjpeg-turbo/libjpeg-turbo.cmake)
 endif()
 
 # PyBind
@@ -323,11 +323,11 @@ list(APPEND 3RDPARTY_INCLUDE_DIRS
      ${flann_INCLUDE_DIRS}
      ${GLEW_INCLUDE_DIRS}
      ${GLFW_INCLUDE_DIRS}
-     ${JPEG_TURBO_INCLUDE_DIRS}
+     #${JPEG_TURBO_INCLUDE_DIRS}
      ${JSONCPP_INCLUDE_DIRS}
      ${liblzf_INCLUDE_DIRS}
      ${tomasakeninemoeller_INCLUDE_DIRS}
-     ${librealsense_INCLUDE_DIRS}
+     #${librealsense_INCLUDE_DIRS}
      ${PNG_INCLUDE_DIRS}
      ${rply_INCLUDE_DIRS}
      ${tinyfiledialogs_INCLUDE_DIRS}
@@ -336,7 +336,7 @@ list(APPEND 3RDPARTY_INCLUDE_DIRS
      ${qhull_INCLUDE_DIRS}
      ${googletest_INCLUDE_DIRS}
      ${fmt_INCLUDE_DIRS}
-     ${k4a_INCLUDE_DIRS}
+     #${k4a_INCLUDE_DIRS}
      ${poisson_INCLUDE_DIRS}
 )
 
@@ -356,14 +356,14 @@ list(APPEND 3RDPARTY_LIBRARY_DIRS
     ${GLFW_LIBRARY_DIRS}
     ${JSONCPP_LIBRARY_DIRS}
     ${PNG_LIBRARY_DIRS}
-    ${JPEG_TURBO_LIBRARY_DIRS}
+    #${JPEG_TURBO_LIBRARY_DIRS}
 )
 
 # set 3RDPARTY_LIBRARIES
 list(APPEND 3RDPARTY_LIBRARIES
      ${GLEW_LIBRARIES}
      ${GLFW_LIBRARIES}
-     ${JPEG_TURBO_LIBRARIES}
+     #${JPEG_TURBO_LIBRARIES}
      ${JSONCPP_LIBRARIES}
      ${PNG_LIBRARIES}
      ${tinyfiledialogs_LIBRARIES}
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 761a6219..70f01715 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -9,7 +9,7 @@ function(ShowAndAbortOnWarning trgt)
 endfunction()
 
 add_subdirectory(Open3D)
-add_subdirectory(Tools)
+#add_subdirectory(Tools)
 if (BUILD_UNIT_TESTS)
     add_subdirectory(UnitTest)
 endif ()
diff --git a/src/Open3D/CMakeLists.txt b/src/Open3D/CMakeLists.txt
index 32186252..3d8831e0 100644
--- a/src/Open3D/CMakeLists.txt
+++ b/src/Open3D/CMakeLists.txt
@@ -12,8 +12,8 @@ add_subdirectory(Integration)
 add_subdirectory(Odometry)
 add_subdirectory(Registration)
 add_subdirectory(Utility)
-add_subdirectory(IO)
-add_subdirectory(Visualization)
+#add_subdirectory(IO)
+#add_subdirectory(Visualization)
 
 # Source group for Visual Studio
 ADD_SOURCE_GROUP(Camera)
@@ -46,8 +46,8 @@ add_library(${CMAKE_PROJECT_NAME}
     $<TARGET_OBJECTS:Odometry>
     $<TARGET_OBJECTS:Registration>
     $<TARGET_OBJECTS:Utility>
-    $<TARGET_OBJECTS:IO>
-    $<TARGET_OBJECTS:Visualization>
+    #$<TARGET_OBJECTS:IO>
+    #$<TARGET_OBJECTS:Visualization>
 )
 ShowAndAbortOnWarning(${CMAKE_PROJECT_NAME})
 
@@ -57,8 +57,8 @@ if (ENABLE_HEADLESS_RENDERING)
     include_directories(${OSMESA_INCLUDE_DIR})
     target_link_libraries(${CMAKE_PROJECT_NAME} ${OSMESA_LIBRARY})
 else ()
-    find_package(OpenGL REQUIRED)
-    target_link_libraries(${CMAKE_PROJECT_NAME} ${OPENGL_LIBRARIES})
+    #find_package(OpenGL REQUIRED)
+    #target_link_libraries(${CMAKE_PROJECT_NAME} ${OPENGL_LIBRARIES})
 endif ()
 
 target_link_libraries(${CMAKE_PROJECT_NAME}
diff --git a/src/Open3D/Open3D.h.in b/src/Open3D/Open3D.h.in
index 6e0adceb..216ec698 100644
--- a/src/Open3D/Open3D.h.in
+++ b/src/Open3D/Open3D.h.in
@@ -68,6 +68,7 @@
 #include "Open3D/Visualization/Utility/DrawGeometry.h"
 #include "Open3D/Visualization/Utility/SelectionPolygon.h"
 #include "Open3D/Visualization/Utility/SelectionPolygonVolume.h"
+/*
 #include "Open3D/Visualization/Visualizer/ViewControl.h"
 #include "Open3D/Visualization/Visualizer/ViewControlWithCustomAnimation.h"
 #include "Open3D/Visualization/Visualizer/ViewControlWithEditing.h"
@@ -76,6 +77,7 @@
 #include "Open3D/Visualization/Visualizer/VisualizerWithEditing.h"
 #include "Open3D/Visualization/Visualizer/VisualizerWithKeyCallback.h"
 #include "Open3D/Visualization/Visualizer/VisualizerWithVertexSelection.h"
+*/
 
 // clang-format off
 @BUILD_AZURE_KINECT_COMMENT@#include "Open3D/IO/Sensor/AzureKinect/AzureKinectRecorder.h"
